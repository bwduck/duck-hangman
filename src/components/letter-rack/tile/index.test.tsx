import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import LetterTile from './index';

describe('<LetterTile/>', () => {
  it('renders and matches the snapshot', () => {
    const handleClick = () => null;
    const wrapper = shallow(
      <LetterTile
        disabled={false}
        letter="A"
        handleLetterTileClick={handleClick}
      />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('calls callback when button is clicked', () => {
    const callback = jest.fn();
    const wrapper = mount(
      <LetterTile
        disabled={false}
        letter="A"
        handleLetterTileClick={callback}
      />
    );
    wrapper.find('button').simulate('click');
    expect(callback).toHaveBeenCalledTimes(1);
  });

  it('should not call callback if button is disabled', () => {
    const callback = jest.fn();
    const wrapper = mount(
      <LetterTile disabled letter="A" handleLetterTileClick={callback} />
    );
    wrapper.find('button').simulate('click');
    expect(callback).toHaveBeenCalledTimes(0);
  });
});
