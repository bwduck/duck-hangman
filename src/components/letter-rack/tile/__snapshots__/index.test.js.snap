// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LetterTile/> renders and matches the snapshot 1`] = `
<button
  className="btn c-letter-tile"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  A
</button>
`;
